src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,443 @@
+                require(noPendingModification, ModificationAlreadyPending());
+
+                // Calculate the new pending diff with this modification
+                mInfo.pendingMagnitudeDelta = int128(uint128(allocation.magnitudes[j])) - int128(uint128(mInfo.currentMagnitude));
+                require(mInfo.pendingMagnitudeDelta != 0, SameMagnitude());
+
+                // Handle deallocation/allocation and modification effect timestamp
+                if (mInfo.pendingMagnitudeDelta < 0) {
+                    // This is a deallocation
+
+                    // 1. Update the effect timestamp for the deallocation
+                    mInfo.effectTimestamp = uint32(block.timestamp) + DEALLOCATION_DELAY;
+                } else if (mInfo.pendingMagnitudeDelta > 0) {
+                    // This is an allocation
+
+                    // 1. add to the encumbered magnitude and make sure it doesn't exceed the total magnitude
+                    uint64 newEncumberedMagnitude = 
+                        encumberedMagnitude[msg.sender][allocation.strategy] 
+                        + uint64(uint128(mInfo.pendingMagnitudeDelta));
+                    require(newEncumberedMagnitude <= totalMagnitude, InsufficientAllocatableMagnitude());
+                    encumberedMagnitude[msg.sender][allocation.strategy] = newEncumberedMagnitude;
+
+                    // 2. Update the effectTimestamp for the allocation
+                    mInfo.effectTimestamp = uint32(block.timestamp) + operatorAllocationDelay;
+                }
+
+                // Add the operatorSet to the modification queue
+                modificationQueue[msg.sender][allocation.strategy].pushBack(operatorSetKey);
+                // Allocate magnitude which will take effect at the `effectTimestamp`
+                _operatorMagnitudeInfo[msg.sender][allocation.strategy][operatorSetKey] = mInfo;
+            }
+        }
+    }
+
+    /**
+     * @notice Called by an AVS to slash an operator for given operatorSetId, list of strategies, and bipsToSlash.
+     * For each given (operator, operatorSetId, strategy) tuple, bipsToSlash
+     * bips of the operatorSet's slashable stake allocation will be slashed
+     *
+     * @param operator the address to slash
+     * @param operatorSetId the ID of the operatorSet the operator is being slashed on behalf of
+     * @param strategies the set of strategies to slash
+     * @param wadToSlash the parts in 1e18 to slash, this will be proportional to the
+     * operator's slashable stake allocation for the operatorSet
+     */
+    function slashOperator(
+        address operator,
+        uint32 operatorSetId,
+        IStrategy[] calldata strategies,
+        uint256 wadToSlash
+    ) external onlyWhenNotPaused(PAUSED_OPERATOR_SLASHING) {
+        require(wadToSlash > 0, InvalidWadToSlash());
```

Comment:
Is there no maximum? Can I submit something over 1e18?

Reply:
7e167f770ec4b99ddffdd9511a60b142fece79ae